### DISCUSSION TITLE
When using GPU computing, the number of CUDA cores is 0, and the GPU utilization is 1%-4%

### DISCUSSION THREAD

**Discussion by XZX2pjq** [DualSPHysics/DualSPHysics#217] [DUALSPHYSICS_V5.2]
Hello developers
I'm using DualSPHysics_v5.2, my GPU version is RTX 4060Ti, and I have CUDA 12.0 installed on Windows, but when calculating, the CUDA kernel shows 0 and the GPU utilisation is very low, is this normal? Or does DualSPHysics_v5.2 not support RTX 4070Ti graphics card calculations?
![屏幕截图 2024-07-23 161407](https://github.com/user-attachments/assets/4cdd7988-2194-4100-8b6f-90916b247504)
![屏幕截图 2024-07-23 161532](https://github.com/user-attachments/assets/e69fbab0-d15c-4edb-8e4d-3728dfe013cd)

**Comment by EnglishA1** [comment]
Hi, 

Can you share some more information? Can you show us the run.out file from one of the simulations and the batch files you are using to run? it could be the case that you have selected the CPU version of the code by accident

Best

Aaron

**Reply by XZX2pjq** [reply]
Hi Aaron
This is my run.out file and batch file, I see that you guessed that I may have selected CPU for simulation, this may be because I am using the CPU for other programs, so my CPU utilisation is up to 85%, can you please help me to see what causes my GPU utilisation to be so low, and whether there is any way to improve it?
[run.out file and batch file.zip](https://github.com/user-attachments/files/16361622/run.out.file.and.batch.file.zip)

Best regrads
XZX2pjq

**Reply by EnglishA1** [reply]
Hi,

Looking at your run.out you are using one of your gpus. From what I can see you are only using a small number of particles and you are outputting at a high frequency so you GPU is probably idle for longer due to the high amount of memory transer with the high saving rate. Can you reduce the rate of saving data and see the effect to every 0.2s instead of every 0.02s. 

Best

Aaron

**Reply by XZX2pjq** [reply]
Hi Aaron

I have adjusted the data saving speed to 0.2s, this is the result after the adjustment, you can see that the GPU utilisation is still very low, I see that you have replied that I am using one GPU, I would like to know if the current v5.2 version can call multiple GPUs for calculations? I remember seeing this question in the forums before and some other developer replied will release a version using multiple GPUs in 2025.
![111](https://github.com/user-attachments/assets/c4aac73a-cbc5-49d0-9011-b24083b1009b)

Best regrads

XZX2pjq

**Reply by EnglishA1** [reply]
Hi XZX,

v5.2 cannot use multi-GPU, it is on the list of future developments that we be available in future releases, but personally I am not sure what date t will be released.

Reducing the output rate has reduced your run time, so you were probably being slowed down by the data transfer. It is also possible that your simulation is too small to utilise the full GPU, as you only 268321 particles. If you want to use more of the GPU then I suggest increasing the resolution of the simulation so you have more particles, but this will take longer to run. 

Best

Aaron

**Comment by jialing95** [comment]
I'm also faced with this problem when trying to use 24 GPUs in v5.2. 
Does v5.4 support multi-GPUs? Can I know the the limit of CPU and GPU in the latest version?

I indeed found that multiple GPUs were recorded in the out file, but how was that if DualSPHysics v5.2 only support single GPU?

**Comment by alexbexe** [comment]
Next version 5.4 will be released in March 2025.
However, the multiGPU approach will be released in v6.0, expected now for 2026.

Regards

**Comment by TAYLOR-QC** [comment]
开发者们好，我是初学者，我遇到了相似的问题，我正在使用DualSPHysics_v5.4，我的GPU版本是RTX 5070Ti，Windows上安装了CUDA 12.3，但计算时，CUDA内核也显示为0，是否和显卡驱动版本有关

**Reply by qq2359586502** [reply]
RTX 5090 too

**Reply by visuttaja** [reply]
Found the cause on my end — not an unsupported-architecture problem, no recompile needed.


System: Windows 11 pro, RTX 5070 Ti + Ryzen 9950X (which has an AMD Radeon integrated GPU). DualSPHysics v5.4.3 prebuilt binary reported "CUDA Cores...: 0 (70 Multiprocessors, 0 CUDA Cores/MP)" for the RTX 5070 Ti — correct name, memory, and compute capability, but 0 cores, and the GPU run just hung at "0%, Calculating..." indefinitely.

Fix: Device Manager → Display adapters → disable the AMD integrated GPU (right-click → Disable device). With only the NVIDIA card enumerated, the GPU solver ran normally — correct core count, RunMode="Pos-Cell - Single-GPU", completed simulation.

Looks like DualSPHysics's device-array indexing gets confused when a second (non-CUDA) GPU is present on the system — name/memory/compute-capability come from one lookup that's fine, but the cores-per-MP lookup lands on the wrong array slot. Worth checking for anyone with a Ryzen APU, laptop with integrated + discrete graphics, etc., before assuming a compile is required.
